TradePlay/src/components/card/add-to-watchlater.jsx

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from "react";
+import { useVideos } from "../../context/videos-context";
+import { inList } from "../../utils/in-list";
+
+const AddToWatchLater = ({ videoData }) => {
+  const { videoState, videoDispatch } = useVideos();
+  const [inWatchlater, setWatchLater] = useState(
+    inList(videoState.watchlater, videoData._id)
+  );
+
+  useEffect(() => {
```

Comment:
Is useEffect necessary here? It's normally used for side-effects.
I am not sure if you are handling any side effects here.

Reply:
Yes I have planned to change that after auth, for now it's temperory. Thanks for feedback.